net/sched/act_skbedit.c

```diff
@@ -50,6 +53,30 @@ static int tcf_skbedit(struct sk_buff *skb, const struct tc_action *a,
 	}
 	if (d->flags & SKBEDIT_F_PTYPE)
 		skb->pkt_type = d->ptype;
+	if (d->flags & SKBEDIT_F_INHERITDSFIELD) {
```

Comment:
1. Move this `if` statement before the if statement `if (d->flags & SKBEDIT_F_PRIORITY)`.

2. Make `if (d->flags & SKBEDIT_F_PRIORITY)` an `else if` of this if statement.